packages/core/http/core-http-router-server-internal/src/router.ts

```diff
@@ -115,6 +115,11 @@ function validOptions(
     );
   }
 
+  // @ts-expect-error to eliminate problems with `security` in the options for route factories abstractions
```

Comment:
Just curious: would you mind sharing an example of where TS does not enforce this for us?

If it's needed, would you mind adding a unit test for this behaviour?

Reply:
This is the one for example https://github.com/elastic/kibana/blob/main/packages/kbn-server-route-repository/src/register_routes.ts#L144 

TS does not check it because of `ServerRoute<string, any, any, any, ServerRouteCreateOptions>` [here](https://github.com/elastic/kibana/blob/main/packages/kbn-server-route-repository/src/register_routes.ts#L46C30-L46C90)

gonna add test 👍